mephisto/abstractions/blueprints/mixins/screen_task_required.py

```diff
@@ -205,13 +205,16 @@ def create_validation_function(
         failed_qualification_name = args.blueprint.block_qualification
 
         def _wrapped_validate(unit):
-            if unit.unit_index >= 0:
+            if args.blueprint.max_screening_units > 0 and unit.unit_index >= 0:
                 return  # We only run validation on the validatable units
-
             agent = unit.get_assigned_agent()
             if agent is None:
                 return  # Cannot validate a unit with no agent
-
+            if (
+                args.blueprint.max_screening_units == 0
+                and agent.get_worker().is_qualified(passed_qualification_name)
```

Comment:
Was slightly confused here if there was a significant difference between checking if it had been granted passing qualification vs checking if it _hadn't_ been granted failing qualification

i.e 
`agent.get_worker().is_qualified(passed_qualification_name)` vs `not agent.get_worker().is_qualified(failed_qualification_name)`

Reply:
Good question - if they've already failed, then they shouldn't be able to get this far, so this is the accurate check.